app/mm-redux/actions/posts.ts

```diff
@@ -214,6 +216,23 @@ export function createPost(post: Post, files: any[] = []) {
             },
         });
 
+        const {message} = post;
+        const RE_UNICODE_EMOJI = emojiRegex();
+        const emojis = message.match(RE_UNICODE_EMOJI);
+        function emojiUnicode(input:string) {
+            if (input.length === 1) {
+                return input.codePointAt(0)?.toString(16);
+            }
+            return input.codePointAt(0)?.toString(16) + '-' + input.codePointAt(1)?.toString(16);
+        }
+        emojis?.map((emoji) => {
+            const unicode = emojiUnicode(emoji);
+            return EmojiIndicesByUnicode.get(unicode || '');
+        })
+        .filter((index) => index !== undefined)
+        .map((index:number) => Emojis[index].aliases[0])
+        .forEach((name) => dispatch(addRecentEmoji(name)));
+
```

Comment:
A couple of important things here:
1. This code  block has a lot of performance implications
   *  iterates over an array 4 times, you could use a reduce to loop through it just once
   * it dispatches an action per found emoji instead of batching or even better would be dispatching just one with an array as its value.

2. Should we add this emojis once the message was posted successfully? (this is more of a UX question, @matthewbirtch thoughts?)

3. I think regardless of the answer for 2, this should be its own isolated action that we could call from the component, specially cause I think this can apply when editing a post.

4. Unit tests have to be added, much easier if you do number 3

Reply:
Hi @enahum 

1. Regarding performance

    - I feel this is more readable than using `.reduce`. I feel this would not cause a performance issue as the number of emojis in a message would be less.
    - I agree with the dispatching part. Will change it. 
2. I think recent reactions should update even if the network request fails.
3.
4.

Reply:
@enahum would https://react-redux.js.org/api/batch help here?

Reply:
1. Yes, readability is important but I rather not iterate over an array 4 times if it can be avoidable.
2. Thanks for your input, appreciate your point of view here.. I do have a slightly different opinion, but it may just be me, I feel adding an emoji from a message to my recent reactions feels weird in the first place, more so if now regardless of being posted or not is populating those recent reactions. That is why I want someone like @matthewbirtch to comment as he is part of the UX team.

Reply:
One more thing, this code bock is not taking into consideration custom emojis which do not have unicode values.

Reply:
If you look one line below we make use of batchAction, that should be ok, but regardless I rather build an array and dispatch just one action instead of batching cause of the way batching works.

Reply:
> 1. Yes, readability is important but I rather not iterate over an array 4 times if it can be avoidable, you say  the  number  of emojis in a message would be less, but we need to consider all cases, people are unpredictable, someone could go about posting a message filled with emojis (seen before), say 200 of them, the now is not so small. (Also I do believe we need to set a limit of the amount we include as recent reactions considering the example I just exposed)
> 2. Thanks for your input, appreciate your point of view here.. I do have a slightly different opinion, but it may just be me, I feel adding an emoji from a message to my recent reactions feels weird in the first place, more so if now regardless of being posted or not is populating those recent reactions. That is why I want someone like @matthewbirtch to comment as he is part of the UX team.

Yeah agreed. Web app has 27 as the limit.

Reply:
@enahum used `for of`, created a new action to update emoji array and implemented maximum limit to recent reactions.

Reply:
> 2\. Should we add this emojis once the message was posted successfully? (this is more of a UX question, @matthewbirtch thoughts?)

I would say it should only be added to the recent reactions list if it was successfully posted. And I think it's a great idea to include reactions I've used in a post to my 'recent reactions' in the emoji picker. I think it's a nice touch.